pkg/controller/deployment/deploment_helper_test.go

```diff
@@ -422,3 +422,369 @@ func TestGetOverrideResourcesFor(t *testing.T) {
 		})
 	}
 }
+
+func TestMergePodScheduling(t *testing.T) {
+	tests := []struct {
+		name               string
+		sourceScheduling   v1alpha1.CertManagerScheduling
+		overrideScheduling v1alpha1.CertManagerScheduling
+		expected           v1alpha1.CertManagerScheduling
+	}{
+		{
+			name: "empty override scheduling doesn't replace source scheduling",
+			sourceScheduling: v1alpha1.CertManagerScheduling{
+				NodeSelector: map[string]string{
+					"nodeLabel": "value",
+				},
+				Tolerations: []corev1.Toleration{
+					{
+						Key:      "toleration",
+						Operator: "Exists",
+						Effect:   "NoSchedule",
+					},
+				},
+			},
+			overrideScheduling: v1alpha1.CertManagerScheduling{},
+			expected: v1alpha1.CertManagerScheduling{
+				NodeSelector: map[string]string{
+					"nodeLabel": "value",
+				},
+				Tolerations: []corev1.Toleration{
+					{
+						Key:      "toleration",
+						Operator: "Exists",
+						Effect:   "NoSchedule",
+					},
+				},
+			},
+		},
+		{
+			name:             "override scheduling replaces empty source scheduling",
+			sourceScheduling: v1alpha1.CertManagerScheduling{},
+			overrideScheduling: v1alpha1.CertManagerScheduling{
+				NodeSelector: map[string]string{
+					"nodeLabel": "value",
+				},
+				Tolerations: []corev1.Toleration{
+					{
+						Key:      "toleration",
+						Operator: "Exists",
+						Effect:   "NoSchedule",
+					},
+				},
+			},
+			expected: v1alpha1.CertManagerScheduling{
+				NodeSelector: map[string]string{
+					"nodeLabel": "value",
+				},
+				Tolerations: []corev1.Toleration{
+					{
+						Key:      "toleration",
+						Operator: "Exists",
+						Effect:   "NoSchedule",
+					},
+				},
+			},
+		},
+		{
+			name: "override scheduling merges with source scheduling",
+			sourceScheduling: v1alpha1.CertManagerScheduling{
+				NodeSelector: map[string]string{
+					"nodeLabel1": "value1",
+				},
+				Tolerations: []corev1.Toleration{
+					{
+						Key:      "toleration1",
+						Operator: "Exists",
+						Effect:   "NoSchedule",
+					},
+				},
+			},
+			overrideScheduling: v1alpha1.CertManagerScheduling{
+				NodeSelector: map[string]string{
+					"nodeLabel2": "value2",
+				},
+				Tolerations: []corev1.Toleration{
+					{
+						Key:      "toleration2",
+						Operator: "Equal",
+						Effect:   "NoSchedule",
+					},
+				},
+			},
+			expected: v1alpha1.CertManagerScheduling{
+				NodeSelector: map[string]string{
+					"nodeLabel1": "value1",
+					"nodeLabel2": "value2",
+				},
+				Tolerations: []corev1.Toleration{
+					{
+						Key:      "toleration1",
+						Operator: "Exists",
+						Effect:   "NoSchedule",
+					},
+					{
+						Key:      "toleration2",
+						Operator: "Equal",
+						Effect:   "NoSchedule",
+					},
+				},
+			},
+		},
+		{
+			name: "override scheduling overrides source scheduling",
```

Comment:
Could you please just add a few more cases (and we can call it complete):

here's the diff that I suggest:
```diff
diff --git a/pkg/controller/deployment/deploment_helper_test.go b/pkg/controller/deployment/deploment_helper_test.go
index 65a0ed7e..6863175d 100644
--- a/pkg/controller/deployment/deploment_helper_test.go
+++ b/pkg/controller/deployment/deploment_helper_test.go
@@ -487,7 +487,7 @@ func TestMergePodScheduling(t *testing.T) {
 			},
 		},
 		{
-			name: "override scheduling merges with source scheduling",
+			name: "override scheduling merges with source scheduling for both nodeSelector and tolerations",
 			sourceScheduling: v1alpha1.CertManagerScheduling{
 				NodeSelector: map[string]string{
 					"nodeLabel1": "value1",
@@ -532,7 +532,7 @@ func TestMergePodScheduling(t *testing.T) {
 			},
 		},
 		{
-			name: "override scheduling overrides source scheduling",
+			name: "override scheduling overrides source scheduling for both nodeSelector and tolerations",
 			sourceScheduling: v1alpha1.CertManagerScheduling{
 				NodeSelector: map[string]string{
 					"nodeLabel": "value1",
@@ -570,6 +570,163 @@ func TestMergePodScheduling(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "override scheduling overrides source scheduling only for nodeSelector",
+			sourceScheduling: v1alpha1.CertManagerScheduling{
+				NodeSelector: map[string]string{
+					"nodeLabel": "value1",
+				},
+				Tolerations: []corev1.Toleration{
+					{
+						Key:      "toleration",
+						Operator: "Equals",
+						Value:    "value",
+						Effect:   "NoSchedule",
+					},
+				}},
+			overrideScheduling: v1alpha1.CertManagerScheduling{
+				NodeSelector: map[string]string{
+					"nodeLabel": "value2",
+				},
+				Tolerations: []corev1.Toleration{},
+			},
+			expected: v1alpha1.CertManagerScheduling{
+				NodeSelector: map[string]string{
+					"nodeLabel": "value2",
+				},
+				Tolerations: []corev1.Toleration{
+					{
+						Key:      "toleration",
+						Operator: "Equals",
+						Value:    "value",
+						Effect:   "NoSchedule",
+					},
+				},
+			},
+		},
+		{
+			name: "override scheduling overrides source scheduling only for tolerations",
+			sourceScheduling: v1alpha1.CertManagerScheduling{
+				NodeSelector: map[string]string{
+					"nodeLabel": "value1",
+				},
+				Tolerations: []corev1.Toleration{
+					{
+						Key:      "toleration",
+						Operator: "Equals",
+						Value:    "value",
+						Effect:   "NoSchedule",
+					},
+				}},
+			overrideScheduling: v1alpha1.CertManagerScheduling{
+				NodeSelector: map[string]string{},
+				Tolerations: []corev1.Toleration{
+					{
+						Key:      "toleration",
+						Operator: "Exists",
+						Effect:   "NoSchedule",
+					},
+				},
+			},
+			expected: v1alpha1.CertManagerScheduling{
+				NodeSelector: map[string]string{
+					"nodeLabel": "value1",
+				},
+				Tolerations: []corev1.Toleration{
+					{
+						Key:      "toleration",
+						Operator: "Exists",
+						Effect:   "NoSchedule",
+					},
+				},
+			},
+		},
+		{
+			name: "override scheduling overrides source nodeSelector and merges tolerations",
+			sourceScheduling: v1alpha1.CertManagerScheduling{
+				NodeSelector: map[string]string{
+					"nodeLabel": "value1",
+				},
+				Tolerations: []corev1.Toleration{
+					{
+						Key:      "toleration1",
+						Operator: "Exists",
+						Effect:   "NoSchedule",
+					},
+				}},
+			overrideScheduling: v1alpha1.CertManagerScheduling{
+				NodeSelector: map[string]string{
+					"nodeLabel": "value2",
+				},
+				Tolerations: []corev1.Toleration{
+					{
+						Key:      "toleration2",
+						Operator: "Equals",
+						Value:    "value",
+						Effect:   "NoSchedule",
+					},
+				},
+			},
+			expected: v1alpha1.CertManagerScheduling{
+				NodeSelector: map[string]string{
+					"nodeLabel": "value2",
+				},
+				Tolerations: []corev1.Toleration{
+					{
+						Key:      "toleration1",
+						Operator: "Exists",
+						Effect:   "NoSchedule",
+					},
+					{
+						Key:      "toleration2",
+						Operator: "Equals",
+						Value:    "value",
+						Effect:   "NoSchedule",
+					},
+				},
+			},
+		},
+		{
+			name: "override scheduling overrides source tolerations and merges nodeSelector",
+			sourceScheduling: v1alpha1.CertManagerScheduling{
+				NodeSelector: map[string]string{
+					"nodeLabel1": "value1",
+				},
+				Tolerations: []corev1.Toleration{
+					{
+						Key:      "toleration",
+						Operator: "Exists",
+						Effect:   "NoSchedule",
+					},
+				}},
+			overrideScheduling: v1alpha1.CertManagerScheduling{
+				NodeSelector: map[string]string{
+					"nodeLabel2": "value2",
+				},
+				Tolerations: []corev1.Toleration{
+					{
+						Key:      "toleration",
+						Operator: "Equals",
+						Value:    "value",
+						Effect:   "NoSchedule",
+					},
+				},
+			},
+			expected: v1alpha1.CertManagerScheduling{
+				NodeSelector: map[string]string{
+					"nodeLabel1": "value1",
+					"nodeLabel2": "value2",
+				},
+				Tolerations: []corev1.Toleration{
+					{
+						Key:      "toleration",
+						Operator: "Equals",
+						Value:    "value",
+						Effect:   "NoSchedule",
+					},
+				},
+			},
+		},
 	}
 
 	for _, tc := range tests {
```

Reply:
Also, the last test case which I added: `name: "override scheduling overrides source tolerations and merges nodeSelector",` fails the expected result: https://github.com/kubernetes/kubernetes/blob/v1.30.1/pkg/util/tolerations/tolerations.go#L73-L107, you'd want to tweak that accordingly to add the superset-ing behaviour (as we can sure trust the `MergeTolerations` func from upstream)..

Reply:
Updated the tests.